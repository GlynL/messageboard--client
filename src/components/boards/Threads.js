import React, { Component } from "react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";

class Threads extends Component {
  componentDidMount() {
    // /:board/threads - all threads api
  }

  renderList() {
    const { board } = this.props;
    return board.threads.map(thread => (
      <li>
        <h2>{thread.name}</h2>
      </li>
    ));
  }

  render() {
    if (
      Object.keys(this.props.board).length === 0 &&
      this.props.board.constructor === Object
    ) {
      return <div>loading...</div>;
    }

    return (
      <div>
        <h1>Threads</h1>
        <div>
          <Link to={`/${this.props.board.name}/new-thread`}>
            Start new thread.
          </Link>
        </div>
        <ul>{this.renderList()}</ul>
      </div>
    );
  }
}

const mapStateToProps = state => ({ board: state.boards.activeBoard });

export default connect(mapStateToProps)(Threads);
