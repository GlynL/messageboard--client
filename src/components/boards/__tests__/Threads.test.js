import React from "react";
import { shallow } from "enzyme";
import { Threads } from "../Threads";

it("matches snapshot initial loading", () => {
  const wrapper = shallow(<Threads board={{}} />);
  expect(wrapper).toMatchSnapshot();
});

describe("with board & thread prop", () => {
  let wrapper, board;
  beforeEach(() => {
    wrapper = shallow(<Threads board={{ name: 'mockboard', _id: 1, threads: [{_id: 1, title: 'mockthread 1', text: 'mock text'}, {_id: 2, title: 'mockthread 2', text: 'mock text'}]}} match={{ params: { board: "mockboard"}}} />)
  })

  it("matches snapshot w/out auth" , () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('matches snapshot w/ auth', () => {
    wrapper.setProps({ authed: 'mockauth'})
    expect(wrapper).toMatchSnapshot();
  })
})