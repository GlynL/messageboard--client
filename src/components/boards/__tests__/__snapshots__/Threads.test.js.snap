// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot initial loading 1`] = `<_default />`;

exports[`with board & thread prop matches snapshot w/ auth 1`] = `
<div
  className="threads"
>
  <h1
    className="threads__title"
  >
    Threads
  </h1>
  <ul
    className="threads__list"
  >
    <li
      className="threads__list-item"
      key="1"
    >
      <Link
        className="threads__list-item-link"
        replace={false}
        to="/mockboard/1"
      >
        <h2
          className="threads__list-item-link-heading"
        >
          mockthread 1
        </h2>
        <p
          className="threads__list-item-link-subheading"
        >
          mock text
        </p>
      </Link>
    </li>
    <li
      className="threads__list-item"
      key="2"
    >
      <Link
        className="threads__list-item-link"
        replace={false}
        to="/mockboard/2"
      >
        <h2
          className="threads__list-item-link-heading"
        >
          mockthread 2
        </h2>
        <p
          className="threads__list-item-link-subheading"
        >
          mock text
        </p>
      </Link>
    </li>
  </ul>
  <div>
    <Link
      className="btn"
      replace={false}
      to="/mockboard/new-thread"
    >
      Start new thread.
    </Link>
  </div>
</div>
`;

exports[`with board & thread prop matches snapshot w/out auth 1`] = `
<div
  className="threads"
>
  <h1
    className="threads__title"
  >
    Threads
  </h1>
  <ul
    className="threads__list"
  >
    <li
      className="threads__list-item"
      key="1"
    >
      <Link
        className="threads__list-item-link"
        replace={false}
        to="/mockboard/1"
      >
        <h2
          className="threads__list-item-link-heading"
        >
          mockthread 1
        </h2>
        <p
          className="threads__list-item-link-subheading"
        >
          mock text
        </p>
      </Link>
    </li>
    <li
      className="threads__list-item"
      key="2"
    >
      <Link
        className="threads__list-item-link"
        replace={false}
        to="/mockboard/2"
      >
        <h2
          className="threads__list-item-link-heading"
        >
          mockthread 2
        </h2>
        <p
          className="threads__list-item-link-subheading"
        >
          mock text
        </p>
      </Link>
    </li>
  </ul>
  <p>
    Sign up to start your own thread.
  </p>
</div>
`;
